### PR TITLE
all lab 2 finished, functional and clean (via grunt)

### PR DESCRIPTION
**Pull request by MTse3**
part 1 has two while loops that are pretty similar, may need to be made into one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,4 @@ comments. Be sure to read everything. Run lab2.js with `node lab2.js`.
 
 
 #### Resources and Collaborators
+mdn and help from Will on some issues on part 3
```
Comment (stephanielingwood) [inline README.md:19]: glad to see you're working with some other students!

File: lab2.js [hunk 2]
```diff
@@ -58,10 +68,19 @@ var hoursSpentInDowington; // TODO: assign me the value of the
...
 
-function hoursToOoze(population, peoplePerHour) {
+Blob.prototype.hoursToOoze = function(population, peoplePerHour) {
   // TODO: implement me based on the instructions above.
   // Be sure to then assign me to the Blob's prototype.
-}
+  this.total = 0;
+  this.rate = peoplePerHour;
+  this.hour = 0;
+  while (this.total < population) {
```
Comment (stephanielingwood) [inline lab2.js:77]: good!

File: lab2.js [hunk 3]
```diff
@@ -84,31 +109,55 @@ var hello = {
...
-function sayHello (sb) {
+SentientBeing.prototype.sayHello = function(sb) {
     // TODO: say hello prints out (console.log's) hello in the
     // language of the speaker, but returns it in the language
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
-
+  console.log(hello[this.languages]);
+  return hello[sb.languages];
```
Comment (stephanielingwood) [inline lab2.js:129]: yes! this is a classic use case for bracket notation.

File: lab2.js [hunk 4]
```diff
@@ -127,24 +176,68 @@ function lastLetterSort(stringArray) {
     // this byLastLetter function is a "compare function"
     // And check out the "comparing strings" section  here:
     // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String
+    var lastA = a.charAt(a.length - 1);
+    var lastB = b.charAt(b.length - 1);
```
Comment (stephanielingwood) [inline lab2.js:180]: good

File: lab2.js [hunk 5]
```diff
@@ -127,24 +176,68 @@ function lastLetterSort(stringArray) {
...
+      return 1;
+    }
+    // a must be equal to b
+    return 0;
   }
   stringArray.sort(byLastLetter);
+  return stringArray;
 }
+var names = ['felix', 'cano', 'cruz', 'seager'];
+var alWest = ['seattle', 'houston', 'los angeles', 'oakland'];
```
Comment (stephanielingwood) [inline lab2.js:194]: and the rangers.

File: lab2.js [hunk 6]
```diff
@@ -127,24 +176,68 @@ function lastLetterSort(stringArray) {
...
+
 }
 
+var arrOfNumbers = [[3, 11, 24, 51], [51, 34, 29, 11], [34, 22, 23, 15]];
+var mariners2015Starters = [[3, 20, 22, 5, 15], [35, 16, 7], [23],
+  [34, 65, 18, 33, 32], [97, 57, 41, 54, 56, 39]];
+
+assert(lastLetterSort(names).toString() === 'cano,seager,felix,cruz',
+  'last letter alphabetically should be o-r-x-z');
+assert(lastLetterSort(alWest).toString() === 'oakland,seattle,houston,los angeles',
```
Comment (stephanielingwood) [inline lab2.js:229]: I like how you figured out a way to test your asserts. toString is useful here.

File: lab2.js (hunk 5 above)
Comment (MTse3) [inline lab2.js:194]: Good catch. I can't believe i forgot about them!

**Pull request merged**